src/main/kotlin/com/openlattice/mechanic/upgrades/V3TimeUseDiaryUpgrade.kt

```diff
@@ -0,0 +1,424 @@
+                    answerNeighborhood.participantId,
+                    setOf(answerNeighborhood.participantESID),
+                    participantFilter,
+                    principals
+                )
+
+                val study = participantNeighbors.filter { it.neighborEntitySet.get().entityTypeId == OL_STUDY_ID }
+                answerNeighborhood.studyId = study.first().neighborId.get() // Only one study should be returned
+
+                answerNeighborhoods.add(answerNeighborhood)
+            }
+
+            // Create Submission objects containing all questions and answers submitted as part of that survey
+            val answersBySubmissionId = answerNeighborhoods.groupBy { it.submissionId }
+            val submissions = createSubmissionsFromAnswerNeighborhoods(answersBySubmissionId)
+
+            // Insert all submissions into postgres
+            insertSubmissions(submissions)
+        }
+        return true
+    }
+
+    /**
+     * Inserts a list of Submission objects with the PreparedStatement bind ordering for a Submission is as follows:
+     * 1) submissionId
+     * 2) organizationId
+     * 3) studyId
+     * 4) participantId
+     * 5) submissionDate
+     * 6) TimeUseDiaryResponse
+     */
+    private fun insertSubmissions(submissions: List<Submission>) {
+        val (hikariConfiguration) = rhizomeConfiguration.datasourceConfigurations["alpr"]!!
+        val hc = HikariConfig(hikariConfiguration)
+        val hds = HikariDataSource(hc)
+        hds.connection.prepareStatement(insertTimeUseDiarySql).use { preparedStatement ->
+            try {
+                for (submission in submissions) {
+                    logger.info("Preparing to insert submission ${submission.submissionId}")
+                    var index = 1
+                    preparedStatement.setObject(index++, submission.submissionId)
+                    preparedStatement.setObject(index++, submission.organizationId)
+                    preparedStatement.setObject(index++, submission.studyId)
+                    preparedStatement.setString(index++, submission.participant)
+                    preparedStatement.setObject(index++, submission.submissionDate, Types.TIMESTAMP_WITH_TIMEZONE)
+                    preparedStatement.setObject(index, objectMapper.writeValueAsString(submission.submission), Types.OTHER)
+                    preparedStatement.addBatch()
+                }
+                logger.info("Executing batch of submission inserts")
+                preparedStatement.executeBatch()
+            } catch (ex: Exception) {
+                logger.error("Error inserting submissions $ex")
+            }
+        }
+    }
+
+    private fun getAllAnswerEntitySets(): Set<Map.Entry<UUID, EntitySet>> {
+        return entitySets.entrySet(
+            Predicates.equal(EntitySetMapstore.ENTITY_TYPE_ID_INDEX, answerEntityTypeId)
+        ) ?: throw NoSuchElementException("No answer entities found.. Aborting")
+    }
+
```

Comment:
This will return all possible answer entity sets, including legacy chronicle_answers entity set. the custom questionnaires feature uses the same model as TUD, and there are a bunch of test questionnaires in the legacy entity sets. Might end up exporting non-TUD related data

I think a better approach would be to grab ONLY v2 organizations that have installed TUD and iterate on those